modules/mono/glue/Managed/Files/Color.cs

```diff
@@ -379,7 +389,303 @@ public string ToHtml(bool include_alpha = true)
+        public static Color Pink { get { return _namedColors["pink"]; } }
+        public static Color Plum { get { return _namedColors["plum"]; } }
+        public static Color PowderBlue { get { return _namedColors["powderblue"]; } }
+        public static Color Purple { get { return _namedColors["purple"]; } }
+        public static Color RebeccaPurple { get { return _namedColors["rebeccapurple"]; } }
+        public static Color Red { get { return _namedColors["red"]; } }
+        public static Color RosyBrown { get { return _namedColors["rosybrown"]; } }
+        public static Color RoyalBlue { get { return _namedColors["royalblue"]; } }
+        public static Color SaddleBrown { get { return _namedColors["saddlebrown"]; } }
+        public static Color Salmon { get { return _namedColors["salmon"]; } }
+        public static Color SandyBrown { get { return _namedColors["sandybrown"]; } }
+        public static Color SeaGreen { get { return _namedColors["seagreen"]; } }
+        public static Color SeaShell { get { return _namedColors["seashell"]; } }
+        public static Color Sienna { get { return _namedColors["sienna"]; } }
+        public static Color Silver { get { return _namedColors["silver"]; } }
+        public static Color SkyBlue { get { return _namedColors["skyblue"]; } }
+        public static Color SlateBlue { get { return _namedColors["slateblue"]; } }
+        public static Color SlateGray { get { return _namedColors["slategray"]; } }
+        public static Color Snow { get { return _namedColors["snow"]; } }
+        public static Color SpringGreen { get { return _namedColors["springgreen"]; } }
+        public static Color SteelBlue { get { return _namedColors["steelblue"]; } }
+        public static Color Tan { get { return _namedColors["tan"]; } }
+        public static Color Teal { get { return _namedColors["teal"]; } }
+        public static Color Thistle { get { return _namedColors["thistle"]; } }
+        public static Color Tomato { get { return _namedColors["tomato"]; } }
+        public static Color Turquoise { get { return _namedColors["turquoise"]; } }
+        public static Color Violet { get { return _namedColors["violet"]; } }
+        public static Color WebGreen { get { return _namedColors["webgreen"]; } }
+        public static Color WebGray { get { return _namedColors["webgray"]; } }
+        public static Color WebMaroon { get { return _namedColors["webmaroon"]; } }
+        public static Color WebPurple { get { return _namedColors["webpurple"]; } }
+        public static Color Wheat { get { return _namedColors["wheat"]; } }
+        public static Color White { get { return _namedColors["white"]; } }
+        public static Color WhiteSmoke { get { return _namedColors["whitesmoke"]; } }
+        public static Color Yellow { get { return _namedColors["yellow"]; } }
+        public static Color YellowGreen { get { return _namedColors["yellowgreen"]; } }
+
```

Comment:
It would be better to do this the opposite way. Create the colors in the properties instead of the dictionary, and make the dictionary's values reference the properties.

Reply:
It may also be better to use a static readonly field rather than a static property, but I don't mind this one.

Reply:
I considered the two scenarios:

  1. define the colors in the properties and then assign them to the dictionary
  2. define the colors in the dictionary and have the properties query them by string index.

With **1** I think that will result in two copies of each color in memory.  However with **2** I think the strings will be interned resulting in less memory usage.